docs/data/material/components/autocomplete/GoogleMaps.tsx

```diff
@@ -138,6 +139,13 @@ export default function GoogleMaps() {
       )}
       renderOption={(props, option) => {
         const matches = option.structured_formatting.main_text_matched_substrings;
+        if (!matches) {
+          return (
+            <ListItem sx={{ color: 'text.secondary', pt: '6px', pb: '6px' }}>
+              No options
+            </ListItem>
+          );
+        }
```

Comment:
I don't understand this logic. If there is an option, there are no "No options". So, if we want to handle no matching substring, I think that we should do it before.

No, Google Maps returns an option, ok, there is no matched substring, but it could still be what the person meant?

<img width="409" alt="Screenshot 2022-12-21 at 01 13 09" src="https://user-images.githubusercontent.com/3165635/208797479-f579bba9-a372-438c-8464-90c4d916ba09.png">

Reply:
Thanks, addressed.